tests/chainer_tests/functions_tests/rnn_tests/test_function_slstm.py

```diff
@@ -29,146 +29,121 @@ def inject_backend_tests(method_names):
+        {'grad_outputs': (True, False)},
+        {'grad_outputs': (False, True)},
+    ], [
+        {'flat': True},
+        {'flat': False},
+    ]
+))
 @testing.fix_random()
-@inject_backend_tests([
-    'test_forward',
-    'test_flat_forward',
-    'test_full_backward',
-    'test_flat_full_backward',
-    'test_no_gc_backward',
-    'test_flat_no_gc_backward',
-    'test_no_gh_backward',
-    'test_flat_no_gh_backward',
-    'test_double_backward'])
-class TestSLSTM(unittest.TestCase):
+@backend.inject_backend_tests(
+    None,
+    # ChainerX tests
+    testing.product({
+        'use_chainerx': [True],
+        'chainerx_device': ['native:0', 'cuda:0'],
+    })
+    # CPU tests
+    + testing.product({
+        'use_cuda': [False],
+        'use_ideep': ['never', 'always'],
+    })
+    # GPU tests
+    + testing.product([
+        [{'use_cuda': True}],
+
+        # Without cuDNN
+        testing.product({
+            'use_cudnn': ['never'],
+        })
+        # With cuDNN
+        + testing.product({
+            'use_cudnn': ['always'],
+            'cudnn_deterministic': [True, False],
+            'autotune': [True, False],
+        })]))
+class TestSLSTM(testing.FunctionTestCase):
+
+    dodge_nondifferentiable = True
 
     def setUp(self):
-        c_prev1 = numpy.random.uniform(-1, 1, (3, 2, 4)).astype(self.dtype)
-        c_prev2 = numpy.random.uniform(-1, 1, (3, 2, 4)).astype(self.dtype)
-        x1 = numpy.random.uniform(-1, 1, (3, 8, 4)).astype(self.dtype)
-        x2 = numpy.random.uniform(-1, 1, (3, 8, 4)).astype(self.dtype)
-
-        gc = numpy.random.uniform(-1, 1, (3, 2, 4)).astype(self.dtype)
-        gh = numpy.random.uniform(-1, 1, (3, 2, 4)).astype(self.dtype)
-
-        ggc_prev1 = numpy.random.uniform(-1, 1, (3, 2, 4)).astype(self.dtype)
-        ggc_prev2 = numpy.random.uniform(-1, 1, (3, 2, 4)).astype(self.dtype)
-        ggx1 = numpy.random.uniform(-1, 1, (3, 8, 4)).astype(self.dtype)
-        ggx2 = numpy.random.uniform(-1, 1, (3, 8, 4)).astype(self.dtype)
-
-        self.inputs = [c_prev1, c_prev2, x1, x2]
-        self.grad_outputs = [gc, gh]
-        self.grad_grad_inputs = [ggc_prev1, ggc_prev2, ggx1, ggx2]
-
-        self.check_forward_options = {}
-        self.check_backward_options = {'dtype': numpy.float64}
-        self.check_double_backward_options = {'dtype': numpy.float64}
         if self.dtype == numpy.float16:
             self.check_forward_options = {'atol': 1e-3, 'rtol': 1e-2}
-            self.check_backward_options = {
-                'dtype': numpy.float64, 'atol': 5e-3, 'rtol': 5e-2}
-            self.check_double_backward_options = {
-                'dtype': numpy.float64, 'atol': 5e-3, 'rtol': 5e-2}
+            self.check_backward_options = {'atol': 5e-3, 'rtol': 5e-2}
+            self.check_double_backward_options = {'atol': 5e-3, 'rtol': 5e-2}
+        if self.grad_outputs[0] is False or self.grad_outputs[1] is False:
+            self.skip_double_backward_test = True
```

Comment:
You don't need to skip the double backward test anymore. Could you remove these two lines above?

Reply:
Done.

Reply:
Hm, it seems like the skip is still needed for old style functions (`Function` and not `FunctionNode`). Could you add the skip back, and maybe leave a TODO comment to remove the skip afterwards?

Reply:
I have made the changes.

Reply:
You only need to skip double backward.